packages/x-date-pickers-pro/src/MultiInputDateRangeField/MultiInputDateRangeField.tsx

```diff
@@ -231,14 +231,16 @@ MultiInputDateRangeField.propTypes = {
   minDate: PropTypes.any,
   /**
    * Callback fired when the value changes.
-   * @template TValue, TError
+   * @template TValue The value type. Will be either the same type as `value` or `null`. Can be in `[start, end]` format in case of range value.
+   * @template TError The validation error type. Will be either `string` or a `null`. Can be in `[start, end]` format in case of range value.
    * @param {TValue} value The new value.
-   * @param {FieldChangeHandlerContext<TError>} The context containing the validation result of the current value.
+   * @param {FieldChangeHandlerContext<TError>} context The context containing the validation result of the current value.
    */
   onChange: PropTypes.func,
   /**
    * Callback fired when the error associated to the current value changes.
-   * @template TValue, TError
+   * @template TValue The value type. Will be either the same type as `value` or `null`. Can be in `[start, end]` format in case of range value.
```

Comment:
Is this rendered anywhere on the doc ?
I just added the `@template` because otherwise Webstorm was complaining that I used non-defined variables in the `JSDoc`.
But if the doc generation is using it, then it's great.

Reply:
In the doc, no, not at the moment.
I forgot to add the result in an IDE.
I'm not 100% sure we need those long descriptions for generic types, especially when they are not visible in IDE, because of type inference and they are not rendered in doc.
My plan was to continue the investigation in how to visually present them to the docs user, because currently the most confusing location is docs. 🙈 
<img width="681" alt="Screenshot 2023-02-07 at 08 45 06" src="https://user-images.githubusercontent.com/4941090/217195677-0edac867-c4f7-4778-9a13-1c2de7041d56.png">

Reply:
OK
If it's rendered correctly in some IDE, then I guess adding those descriptions is a nice improvement.
I do agree that we must find a way to render it on the doc.